src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/builders/ClassBuilder.java

```diff
@@ -467,7 +450,10 @@ private void setRecordDocumentation(TypeElement elem) {
             }
         }
 
-        for (VariableElement ve : utils.getFields(elem)) {
+        var fields = utils.isSerializable(elem)
+                ? utils.getFieldsUnfiltered(elem)
+                : utils.getFields(elem);
```

Comment:
I guess the fields are filtered for inclusion later on, otherwise they would be included in the record's class docuentation as well?

Reply:
Mostly, I was just trying to minimize changes to existing code for non-serializable classes.

This part of the code is just about synthesizing doc comments that may be required later on.  If the class is not serializable, `getFields` will take command-line options like `-private` into account and only generate comments that will be required.  If the class is serializable, we want to ignore the default access control and always generate comments because we know they will be required on the Serialized Form page.